pkg/oci/utils.go

```diff
@@ -161,61 +197,94 @@ func networkConfig(ocispec CompatOCISpec) (vc.NetworkConfig, error) {
 	return netConf, nil
 }
 
-// PodConfig converts an OCI compatible runtime configuration file
-// to a virtcontainers pod configuration structure.
-func PodConfig(runtime RuntimeConfig, bundlePath, cid, console string) (*vc.PodConfig, *CompatOCISpec, error) {
-	configPath := filepath.Join(bundlePath, "config.json")
+// getConfigPath returns the full config path from the bundle
+// path provided.
+func getConfigPath(bundlePath string) string {
+	return filepath.Join(bundlePath, "config.json")
+}
+
+// ParseConfigJSON unmarshals the config.json file.
+func ParseConfigJSON(bundlePath string) (CompatOCISpec, error) {
+	configPath := getConfigPath(bundlePath)
 	ociLog.Debugf("converting %s", configPath)
 
 	configByte, err := ioutil.ReadFile(configPath)
 	if err != nil {
-		return nil, nil, err
+		return CompatOCISpec{}, err
 	}
 
 	var ocispec CompatOCISpec
-	if err = json.Unmarshal(configByte, &ocispec); err != nil {
-		return nil, nil, err
+	if err := json.Unmarshal(configByte, &ocispec); err != nil {
+		return CompatOCISpec{}, err
 	}
 
-	rootfs := ocispec.Root.Path
-	if !filepath.IsAbs(rootfs) {
-		rootfs = filepath.Join(bundlePath, ocispec.Root.Path)
+	return ocispec, nil
+}
+
+// GetContainerType determines which type of container matches the annotations
+// table provided.
+func GetContainerType(annotations map[string]string) (vc.ContainerType, error) {
```

Comment:
I thought we talked in private about this one, and we agreed that we cannot move this as a virtcontainers method because that means virtcontainers would be aware about all annotations. And we decided that we wanted virtcontainers to be annotations agnostic.